### PR TITLE
Helpers for Ownable contract

### PR DESCRIPTION
**Pull request by nkuba**
Function to easily transfer ownership of an Ownable contract to a specific address.
If the current owner is already a desired address it will not execute a call to the contract.

Depends on: https://github.com/keep-network/hardhat-helpers/pull/3

### CODE REVIEW

**Comment by mhluongo** [comment]
Parachuting out of nowhere with questions!

1. Why is the `dist/` dir included in this repo? Would it make sense to use something like `tsdx` and have `dist` created as part of the build / publish?
2. It feels like you're writing TS that can be auto-generated with something like `typechain`, which we use elsewhere... what's the rationale for declaring your own `Ownable` class?

**Comment by nkuba** [comment]
>     1. Why is the `dist/` dir included in this repo? Would it make sense to use something like `tsdx` and have `dist` created as part of the build / publish?

Thanks for pointing that to me! Obviously, we don't need to push the `dist` directory (this was probably something that I brought from developing GH actions, where `dist` dir had to be pushed). It was enough to define `prepare` script that will be invoked on `yarn install` in the package using this plugin (see: https://github.com/keep-network/hardhat-helpers/pull/3).

>     2. It feels like you're writing TS that can be auto-generated with something like `typechain`, which we use elsewhere... what's the rationale for declaring your own `Ownable` class?

The `Ownable` class is a helper that implements a specific logic that we need for deployments scripts. It's not directly calling the contracts. Also for calls to the contract, we use functions exposed by `hardhat-deploy` plugin.

File: src/Ownable.ts [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  constructor(hre: HardhatRuntimeEnvironment) {
+    this.hre = hre
+  }
+
+  /**
+   * Transfers ownership of an Ownable contract to a specific address.
+   *
+   * @param {string} contractName Deployed Contract Name
+   * @param {string} newOwnerAddress New Owner Address
+   * @param {from} from Address to send transaction from.
```
Comment (r-czajkowski) [inline src/Ownable.ts:15]: Should be `string` instead of `from`.

File: src/Ownable.ts [hunk 2]
```diff
@@ -0,0 +1,42 @@
+import type { HardhatRuntimeEnvironment } from "hardhat/types"
```
Comment (r-czajkowski) [inline src/Ownable.ts:1]: VSC shows me an error here and in other places where we use `import type`. Getting rid of a `type` did the trick but not sure which approach is better/correct.

obraz

File: tsconfig.json [hunk 3]
```diff
@@ -12,6 +12,6 @@
     "esModuleInterop": true,
     "resolveJsonModule": true
   },
-  "exclude": ["dist", "node_modules"],
+  "exclude": ["dist"],
```
Comment (r-czajkowski) [inline tsconfig.json:15]: Was this change intentional ?

**Review by r-czajkowski — CHANGES_REQUESTED** [review]

File: src/Ownable.ts (hunk 1 above)
Comment (nkuba) [reply src/Ownable.ts]: fa5c8488edba0c6dba0423d0848bbc5e0c117b52

File: src/Ownable.ts (hunk 2 above)
Comment (nkuba) [reply src/Ownable.ts:1]: Please make sure you're on [TypeScript 3.8](https://devblogs.microsoft.com/typescript/announcing-typescript-3-8-beta/#type-only-imports-exports) or higher. It looks fine in my VSC.

File: tsconfig.json (hunk 3 above)
Comment (nkuba) [reply tsconfig.json]: I'm not sure, it doesn't look like we should make this change. I reverted it in 1cd4f466280fdb8965009481aea44b791b15c13e.

File: src/types.ts [hunk 4]
```diff
@@ -1,5 +1,7 @@
 import type { Address } from "./Address"
+import { Ownable } from "./Ownable"
```
Comment (r-czajkowski) [inline src/types.ts:2]:
```suggestion
import type { Ownable } from "./Ownable"
```
Because `Ownable` is used only for type annotations and declarations.

**Review by r-czajkowski — APPROVED** [review]
Left a tiny comment and we can merge.
